lib/facilities/ppms_client.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+
+module Facilities
+  # Core class responsible for api interface operations
+  # Web forum and documentation (latest version of the ICD) located at:
+  # https://vaww.oed.portal.va.gov/pm/iehr/vista_evolution/RA/CCP_PPMS/PPMS_DWS/SitePages/Home.aspx
+  # Dev swagger site for testing endpoints
+  # https://dev.dws.ppms.va.gov/swagger
+  class PPMSClient < Common::Client::Base
+    configuration Facilities::PPMSConfiguration
+
+    # https://dev.dws.ppms.va.gov/swagger/ui/index#!/GlobalFunctions/GlobalFunctions_ProviderLocator
+    def provider_locator(params)
+      qparams = build_params(params)
+      response = perform(:get, 'v1.0/ProviderLocator?', qparams)
+      bbox_num = params[:bbox].map { |x| Float(x) }
+      response.body.select! do |provider|
+        provider['Latitude'] > bbox_num[1] && provider['Latitude'] < bbox_num[3] &&
+          provider['Longitude'] > bbox_num[0] && provider['Longitude'] < bbox_num[2]
+      end
+      response.body
+    end
+
+    # https://dev.dws.ppms.va.gov/swagger/ui/index#!/Providers/Providers_Get_0
+    def provider_info(identifier)
+      qparams = { :$expand => 'ProviderSpecialties' }
+      response = perform(:get, "v1.0/Providers(#{identifier})?", qparams)
+      response.body[0]
+    end
+
+    # https://dev.dws.ppms.va.gov/swagger/ui/index#!/Specialties/Specialties_Get_0
+    def specialties
+      response = perform(:get, 'v1.0/Specialties', {})
+      response.body
+    end
+
+    def build_params(params)
+      bbox_num = params[:bbox].map { |x| Float(x) }
```

Comment:
any chance params[:bbox] values can't be cast as Floats? The facilities controller catches ArgumentErrors in those cases https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/controllers/facilities_controller.rb#L11

Reply:
Currently (in my main development branch) this function is being called from the context of the search endpoint in the api meaning that it will go through a validation check before it gets to this call. I can add another check here if you think it makes sense/in case it's ever necessary to use this method in a different context

Reply:
@SimonSchama ah ok, if it's gone through the validation check once then that's cool 👍